be2-scala/src/main/scala/ch/epfl/pop/decentralized/ConnectionMediator.scala

```diff
@@ -37,28 +38,22 @@ final case class ConnectionMediator(
             messageRegistry,
             monitorRef,
             self,
-            isServer = true
+            isServer = true,
+            initGreetServer = true
           )
         )
       )
 
-    case ConnectionMediator.NewServerConnected(serverRef) =>
-      log.info("Received new server")
-      if (serverSet.isEmpty)
-        monitorRef ! Monitor.AtLeastOneServerConnected
-
-      serverSet += serverRef
-
     case ConnectionMediator.ServerLeft(serverRef) =>
```

Comment:
I see that in line 67, you removed the `ConnectionMediator.`, because I guess it is not necessary to put it. For consistency, you should also remove this one, or put it in both cases.